integreat_cms/core/management/commands/fix_internal_links.py

```diff
@@ -138,50 +102,37 @@ def handle(self, *args, region_slug, username, commit, **options):
         region = get_region(region_slug) if region_slug else None
         user = get_user(username) if username else None
 
-        # Get all broken internal urls
-        broken_internal_urls = [
-            url
-            for url in Url.objects.all()
-            if url.type == "internal" and not url.status
-        ]
-
-        for url in broken_internal_urls:
-            (
-                region_slug,
-                language_slug,
-                content_type,
-                translation_slug,
-            ) = parse_internal_url(url)
-
-            if content_type in [settings.IMPRINT_SLUG, "news", "offers"]:
-                # Only links to pages, events or locations can be fixed automatically
-                continue
-
-            # Find a content object by translation url
-            objects = (
-                get_content_model(content_type)
-                .objects.filter(
-                    translations__slug=slugify(translation_slug, allow_unicode=True),
-                    translations__language__slug=language_slug,
-                    region__slug=region_slug,
-                )
-                .distinct()
+        for url in Url.objects.all():
```

Comment:
This feels like a lot of overhead... why not filtering out all potential internal urls first? We have ~80k URL objects in the database and only ~8k of them are internal.

Reply:
With this pr, the fix internal links command now also fixes short urls which point to the wrong language. Unfortunately they are not considered internal, since they have the backend domain, not the app domain.